cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/FormRecognizerV3.scala

```diff
@@ -0,0 +1,92 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.cognitive
+
+import com.microsoft.azure.synapse.ml.logging.BasicLogging
+import org.apache.http.entity.{AbstractHttpEntity, ByteArrayEntity, ContentType, StringEntity}
+import org.apache.spark.ml.ComplexParamsReadable
+import org.apache.spark.ml.param.ServiceParam
+import org.apache.spark.ml.util.Identifiable
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.types.DataType
+import spray.json.DefaultJsonProtocol._
+import spray.json._
+
+trait HasPrebuiltModelID extends HasServiceParams {
+  // TODO: Add back custom modelId support after cog service team's fix
+  // Currently custom modelId generated from the old version can't be used by V3
+  val prebuiltModelId = new ServiceParam[String](this, "prebuiltModelId",
+    "Prebuilt Model identifier for Form Recognizer V3.0, supported modelId: prebuilt-layout," +
+      "prebuilt-document, prebuilt-businessCard, prebuilt-idDocument, prebuilt-invoice, prebuilt-receipt," +
+      "or your custom modelId", {
+      case Left(s) => Set("prebuilt-layout", "prebuilt-document", "prebuilt-businessCard",
+        "prebuilt-idDocument", "prebuilt-invoice", "prebuilt-receipt")(s)
+      case Right(_) => true
+    }, isRequired = true)
+
+  def setPrebuiltModelId(v: String): this.type = setScalarParam(prebuiltModelId, v)
+
+  def setPrebuiltModelIdCol(v: String): this.type = setVectorParam(prebuiltModelId, v)
+
+  def getPrebuiltModelId: String = getScalarParam(prebuiltModelId)
+
+  def getPrebuiltModelIdCol: String = getVectorParam(prebuiltModelId)
+}
+
+object AnalyzeDocumentV3 extends ComplexParamsReadable[AnalyzeDocumentV3]
+
+class AnalyzeDocumentV3(override val uid: String) extends CognitiveServicesBaseNoHandler(uid)
```

Comment:
are we able to call this AnalyzeDocument and rename previous to AnalyzeDocumentV2? Thats what we did for TA update